crates/sui-core/src/safe_client.rs

```diff
@@ -18,23 +20,102 @@ use sui_types::{
 use tap::TapFallible;
 use tracing::info;
 
+/// Prometheus metrics which can be displayed in Grafana, queried and alerted on
+#[derive(Clone, Debug)]
+pub struct SafeClientMetrics {
+    pub total_requests_by_address_method: IntCounterVec,
+    pub total_responses_by_address_method: IntCounterVec,
+}
+
+impl SafeClientMetrics {
+    pub fn new(registry: &prometheus::Registry) -> Self {
+        Self {
+            total_requests_by_address_method: register_int_counter_vec_with_registry!(
+                "safe_client_total_requests_by_address_method",
+                "Total requests to validators group by address and method",
+                &["address", "method"],
+                registry,
+            )
+            .unwrap(),
+            total_responses_by_address_method: register_int_counter_vec_with_registry!(
+                "safe_client_total_responses_by_address_method",
+                "Total good (OK) responses from validators group by address and method",
+                &["address", "method"],
+                registry,
+            )
+            .unwrap(),
+        }
+    }
+
+    pub fn new_for_tests() -> Self {
+        let registry = prometheus::Registry::new();
+        Self::new(&registry)
+    }
+}
+
 #[derive(Clone)]
 pub struct SafeClient<C> {
     authority_client: C,
     committee: Committee,
     address: AuthorityPublicKeyBytes,
+
```

Comment:
Maybe a comment here to say the meaning of these can be found in the definition of `SafeClientMetrics::new` and that they are specialised metrics to this client / particular method.